src/libsyntax/feature_gate.rs

```diff
@@ -792,7 +792,7 @@ pub enum GateIssue {
 }
 
 pub fn emit_feature_err(sess: &ParseSess, feature: &str, span: Span, issue: GateIssue,
-                        explain: &str) {
+                        explain: &str, help_note: Option<&str>) {
```

Comment:
I would instead make a version of this function (say, `feature_err`) that returns the `err` instead of calling `err.emit()`, and have this function just do `feature_err(...).emit()`. Then the existing callers of this function don't need to change for the `None` argument, and the `stmt_expr_attributes` one can do `let err = feature_err(...); err.help(...); err.emit();`.